docs/PROPOSAL-backup-location-layout.md

```diff
@@ -0,0 +1,116 @@
+# Overview
+
+This document is a proposal for a revised layout for backup storage locations.
+
+# Current Layout
+
+Backup storage locations are structured as follows:
+```
+<bucket>/
+  <prefix>/ (optional)
+    backup-1/
+      ark-backup.json
+      backup-1.tar.gz
+      backup-1-logs.tar.gz
+      restore-of-backup-1-logs.gz
+      restore-of-backup-1-results.gz
+    backup-2/
+      ark-backup.json
+      backup-2.tar.gz
+      backup-2-logs.tar.gz
+      restore-of-backup-2-logs.gz
+      restore-of-backup-2-results.gz
+    ...
+  <prefix>/
+    ...
+```
+
+Ark's restic integration requires a separate "root" location, i.e. a different bucket + prefix.  Restic locations are structured as follows:
+```
+<bucket>/
+  <prefix>/ (optional)
+    namespace-1/
+      config
+      data/
+      index/
+      keys/
+      locks/
+      snapshots/
+    namespace-2/
+      config
+      data/
+      index/
+      keys/
+      locks/
+      snapshots/
+    ...
+```
+
+# Revised Layout
+
+The revised layout adds top-level "directories" within the root of the backup storage location to organize the contents.
+The initial set of top-level directories is:
```

Comment:
I would add how the concept of ownership and read/write access factors into each directory here.

e.g. I expect only the owner to write to `backups` and `restic`. All other clusters could read all data, and potentially write to `restores` and possibly `metadata`, such as when claiming ownership after a migration.

Reply:
Will do. I think it probably makes the most sense for restore logs/results of backups not owned by the target cluster to just go into the target cluster's default location under restores/ - WDYT?

Reply:
Added some info on this.  Gonna punt on deciding where restore logs go 'til next release.